granary/tests/test_bluesky.py

```diff
@@ -823,3 +917,143 @@ def test_get_activities_self_user_id(self, mock_get):
+        json=None,
+        headers={
+          'Authorization': 'Bearer towkin',
+          'Content-Type': 'application/json',
+          'User-Agent': util.user_agent,
+        },
+    )
+
+  @patch('requests.get')
+  def test_get_activities_with_reposts(self, mock_get):
+    mock_get.side_effect = [
+      requests_response({
+        'cursor': 'timestamp::cid',
+        'feed': [POST_FEED_VIEW_WITH_REPOSTS_BSKY],
+      }),
+      requests_response({
+        'cursor': 'timestamp::cid',
+        'uri': 'at://did/app.bsky.feed.post/tid',
+        'repostedBy': [ACTOR_PROFILE_VIEW_BSKY]
+      })
+    ]
+
+    self.assert_equals(
+      [POST_AUTHOR_PROFILE_WITH_REPOSTS_AS],
+      self.bs.get_activities(fetch_shares=True)
+    )
+    mock_get.assert_any_call(
+        'https://bsky.social/xrpc/app.bsky.feed.getTimeline',
+        json=None,
+        headers={
+          'Authorization': 'Bearer towkin',
+          'Content-Type': 'application/json',
+          'User-Agent': util.user_agent,
+        },
+    )
+    mock_get.assert_any_call(
+        'https://bsky.social/xrpc/app.bsky.feed.getRepostedBy?uri=at%3A%2F%2Fdid%2Fapp.bsky.feed.post%2Ftid',
+        json=None,
+        headers={
+          'Authorization': 'Bearer towkin',
+          'Content-Type': 'application/json',
+          'User-Agent': util.user_agent,
+        },
+    )
+
+  @patch('requests.get')
+  def test_get_activities_include_shares(self, mock_get):
+    mock_get.return_value = requests_response({
+      'cursor': 'timestamp::cid',
+      'feed': [POST_FEED_VIEW_BSKY, REPOST_BSKY_FEED_VIEW_POST],
+    })
+
+    self.assert_equals(
+      [POST_AUTHOR_PROFILE_AS, REPOST_PROFILE_AS],
+      self.bs.get_activities(include_shares=True)
+    )
+
+  @patch('requests.get')
+  def test_get_activities_with_replies(self, mock_get):
+    mock_get.side_effect = [
+      requests_response({
+        'cursor': 'timestamp::cid',
+        'feed': [POST_FEED_VIEW_WITH_REPLIES_BSKY],
+      }),
+      requests_response({
+        'thread': THREAD_BSKY,
+      })
+    ]
+
+    self.assert_equals([THREAD_AS],
+                       self.bs.get_activities(fetch_replies=True))
+    mock_get.assert_any_call(
+        'https://bsky.social/xrpc/app.bsky.feed.getTimeline',
+        json=None,
+        headers={
+          'Authorization': 'Bearer towkin',
+          'Content-Type': 'application/json',
+          'User-Agent': util.user_agent,
+        },
+    )
+    mock_get.assert_any_call(
+        'https://bsky.social/xrpc/app.bsky.feed.getPostThread?uri=at%3A%2F%2Fdid%2Fapp.bsky.feed.post%2Ftid',
+        json=None,
+        headers={
+          'Authorization': 'Bearer towkin',
+          'Content-Type': 'application/json',
+          'User-Agent': util.user_agent,
+        },
+    )
+
+  @patch('requests.get')
+  def test_get_comment(self, mock_get):
+    mock_get.return_value = requests_response({
+      '$type': 'app.bsky.feed.defs#threadViewPost',
```

Comment:
I think you can drop this and `replies`, they're supposed to be inside `thread` instead of outside it, like in `test_get_activities_with_replies`.

Reply:
good spot!